### PR TITLE
Ender 3 S1 configuration

### PR DESCRIPTION
**Pull request by Sebazzz**
### Requirements

Adds an example configuration for the Ender 3 S1 based on latest bugfix-2.0.x.

### Description

- The LCD has slight graphical glitches but is very usable. 
- I was conservative with the bed X/Y size and kept 220mm from the Creality firmware, though 230 is fine to use.
- Due to the lack of Marlin firmware (which is due to the lack of development tools for this unknown display) we need to refer to the original Creality DWIN icon indices and do depend on the latest firmware being flashed.

### Benefits

Enable Creality Ender 3 S1 support

### Related Issues

https://github.com/MarlinFirmware/Marlin/pull/23620

### CODE REVIEW

File: config/examples/Creality/Ender-3 S1/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Therefore, to flash the compiled firmware binary onto the board you must give the "`firmware.bin`" file on the SD card a unique name, different from the name of the previous firmware file, or you will be greeted with a blank screen on the next boot.
+
+## Updating the Display
+
+Currently there is no Marlin display firmware available, so flash the latest official firmware to the display. There are some minor graphical glitches but the display is very usable.
+
+Download the latest Ender 3 S1 firmware [from www.creality.com/download](https://www.creality.com/download) and flash as follows:
+
+- Copy the `private` folder to a 4096k formatted DAT32 microSD card and insert the card into the slot on the back of the display unit.
```
Comment (CRCinAU) [inline config/examples/Creality/Ender-3 S1/README.md]: Typo.... DAT32 -> FAT32

File: config/examples/Creality/Ender-3 S1/Configuration.h [hunk 2]
```diff
@@ -0,0 +1,3006 @@
...
+ * :[-1, 0, 1, 2, 3, 4, 5, 6, 7]
+ */
+//#define SERIAL_PORT_3 1
+//#define BAUDRATE_3 250000   // Enable to override BAUDRATE
+
+// Enable the Bluetooth serial interface on AT90USB devices
+//#define BLUETOOTH
+
+// Name displayed in the LCD "Ready" message and Info menu
+#define CUSTOM_MACHINE_NAME "Ender 3 S1 Sebazzz"
```
Comment (CRCinAU) [inline config/examples/Creality/Ender-3 S1/Configuration.h]: I believe we tend to discourage non-model parts in here - it should likely be just "Ender 3 S1"

Comment (The-EG) [reply config/examples/Creality/Ender-3 S1/Configuration.h]: Author name would be more appropriate in `STRING_CONFIG_H_AUTHOR` if desired, something like
```c++
#define STRING_CONFIG_H_AUTHOR "(Sebazzz, Ender 3 S1)"
```

File: config/examples/Creality/Ender-3 S1/Configuration.h [hunk 3]
```diff
@@ -0,0 +1,3006 @@
...
+#if ENABLED(PIDTEMPBED)
+  //#define MIN_BED_POWER 0
+  //#define PID_BED_DEBUG // Sends debug data to the serial port.
+
+  // 120V 250W silicone heater into 4mm borosilicate (MendelMax 1.5+)
+  // from FOPDT model - kp=.39 Tp=405 Tdead=66, Tc set to 79.2, aggressive factor of .15 (vs .1, 1, 10)
+  #define DEFAULT_bedKp 462.10
+  #define DEFAULT_bedKi 85.47
+  #define DEFAULT_bedKd 624.59
+
```
Comment (CRCinAU) [inline config/examples/Creality/Ender-3 S1/Configuration.h:662]: These values seem to be a lot different than other printers. Does the S1 have a different heatbed than others?

ie: the Ender 3 config is:
```
  #define DEFAULT_bedKp 10.00
  #define DEFAULT_bedKi .023
  #define DEFAULT_bedKd 305.4
```

**Review by CRCinAU — CHANGES_REQUESTED** [review]
The rest mostly looks ok - but don't have this specific printer to test a build on.

Address the few issues mentioned and it should be good to go.

File: config/examples/Creality/Ender-3 S1/Configuration.h (hunk 2 above)
Comment (Sebazzz) [reply config/examples/Creality/Ender-3 S1/Configuration.h]: Oops, left this accidently in after porting my own personal config!

File: config/examples/Creality/Ender-3 S1/Configuration.h (hunk 3 above)
Comment (Sebazzz) [reply config/examples/Creality/Ender-3 S1/Configuration.h:662]: No, but this is the PID tuned result. For the S1 this matters more that the original due to the PID sheet.

File: config/examples/Creality/Ender-3 S1/README.md (hunk 1 above)
Comment (Sebazzz) [reply config/examples/Creality/Ender-3 S1/README.md]: Fixed

File: config/examples/Creality/Ender-3 S1/Configuration.h (hunk 2 above)
Comment (Sebazzz) [reply config/examples/Creality/Ender-3 S1/Configuration.h]: Fixed

**Comment by Sebazzz** [comment]
Addressed your comments.

**Pull request merged**